x/nexus/types/genesis.go

```diff
@@ -91,6 +97,18 @@ func (m GenesisState) Validate() error {
 		}
 	}
 
+	for _, rateLimit := range m.RateLimits {
```

Comment:
Should also make sure that `.RateLimits` and `.TransferRates` don't contain duplicate for however they are keyed.